server/src/main/java/org/opentosca/toscana/plugins/lifecycle/LifecycleAwarePlugin.java

```diff
@@ -0,0 +1,167 @@
+                    " because the model check has failed!");
+            }
+        }));
+
+        addPostValidatePhases(executionTasks);
+
+        //Transformation phases
+        executionTasks.add(new ExecutionPhase<>("prepare", TransformationLifecycle::prepare));
+
+        addPostPreparePhases(executionTasks);
+
+        executionTasks.add(new ExecutionPhase<>("transformation", TransformationLifecycle::transform));
+
+        addPostTransformPhases(executionTasks);
+
+        executionTasks.add(new ExecutionPhase<>("cleanup", TransformationLifecycle::cleanup));
+
+        //Make list immutable
+        executionTasks = Collections.unmodifiableList(executionTasks);
+    }
+
+    /**
+     This is a method that allows the Plugin Developer to add additional execution phases
+     <p>
+     Can be implemented by the implementing plugins to add tasks that should be executed after the
+     Validation phase and before the prepare phase
+
+     @param executionTasks Add the Phases in the order that they should executed (first one gets inserted first...)
+     */
+    protected void addPostValidatePhases(List<ExecutionPhase<LifecycleT>> executionTasks) {
+        //Can be implemented by the implementing plugins to add tasks that should be executed after the 
+        //Validation phase and before the prepare phase
+    }
+
+    /**
+     This is a method that allows the Plugin Developer to add additional execution phases
+     <p>
+     Can be implemented by the implementing plugins to add tasks that should be executed after the
+     prepare phase and before the transform phase
+
+     @param executionTasks Add the Phases in the order that they should executed (first one gets inserted first...)
+     */
+    protected void addPostPreparePhases(List<ExecutionPhase<LifecycleT>> executionTasks) {
+        //Can be implemented by the implementing plugins to add tasks that should be executed after the 
+        //prepare phase and before the transform phase
+    }
+
+    /**
+     This is a method that allows the Plugin Developer to add additional execution phases
+     <p>
+     Can be implemented by the implementing plugins to add tasks that should be executed after the
+     Transform phase and before the cleanup phase
+
+     @param executionTasks Add the Phases in the order that they should executed (first one gets inserted first...)
+     */
+    protected void addPostTransformPhases(List<ExecutionPhase<LifecycleT>> executionTasks) {
+        //Can be implemented by the implementing plugins to add tasks that should be executed after the 
+        //Transform phase and before the cleanup phase
+    }
+
+    /**
+     Performs the executon of the phases in the order that has been defined during the construction of the object
+
+     @param context context for the transformation
+     */
+    @Override
+    public void transform(TransformationContext context) throws Exception {
+        Logger logger = context.getLogger(getClass().getName());
```

Comment:
context.getLogger(getClass()) should be sufficient.

Reply:
This is required to mock the transformation context when testing. Mockito does not provide a `anyClass()` matcher

Reply:
doesn't `any()` work?

Reply:
Nope.
Found a workaround though:

``(Class<?) any(Class.class)` works